### DISCUSSION TITLE
Is this project active?

### DISCUSSION THREAD

**Discussion by BEllis** [josephg/node-foundationdb#65] [General]
I've not seen any commits since Nov 2021? Is this project still active? I'd like to contribute if possible, putting my changes in my fork for now.

**Comment by josephg** [comment]
Ok sure. I've been maintaining this as issues come up, but I'm not using it. So I haven't been actively working on it. Its in maintenance mode (and probably will be in the indefinite future while I focus my attention on other projects).

Forking is probably a good idea if you want to make nontrivial changes to the codebase.

**Comment by josephg** [comment]
FYI, changing the license of someone else's code without asking is a bit rude. I also don't think you can do that legally, because the old license includes this text:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

As I read it, removing the old copyright notice is literally the only thing you can do which violates the previous license.

At this point (I think) all the code is mine, though the directory layer code is basically a straight port of the equivalent code in the python library. I'd be happy to dual license my code under the apache-2 license if you ask. But it feels weird to not be asked, and to accidentally find a mark of my own authorship stripped off my own work.

(I am not a lawyer, this is not legal advice)

**Reply by BEllis** [reply]
Sorry, didn't mean to offend or break any licenses. I'll change the license back. I was just thinking since foundation db and it's binaries are all under apache 2.0 it'd make sense for this project to be also.

My fork is mostly for me to play around with some ideas and to gain an understanding of how it all fits together. If I raise PRs I'll likely do it a piece at a time and not my entire fork.

Some bits from my fork thus far that might be useful, let me know if you think they'd be useful and I'll put together some PRs.

- Moved all distributable items into dist directory and all source items in src directory (not sure about the src directory bit, its more of a maven style project structure while npm modules tend to put everything in lib)
- Added docker container to tests to auto spin up a new foundation db instance on start up.

I've been trying to follow your conversations and work around the multi-versioning and prebuilt binaries that I've seen online. Did you ever reach a conclusion?

I took a look at the node-oracledb package (https://oracle.github.io/node-oracledb/INSTALL.html#quickstart) and they seem to have taken the approach of, bring-your-own client libraries (I suspect most likely because oracle licensing doesn't let you distribute the libraries), but I think having an optional multi-version package with all the versions prebuilt would be a nice approach? If we could change the library to support multiple versions at runtime and then it's up to the implementer to specify what version should be used to connect to a cluster.
